packages/core/src/ui/UIUtil.ts

```
@@ -0,0 +1,114 @@
+      parents[i]._unRegisterModifyListener(element._onEntityModify);
+    }
+    parents.length = 0;
+  }
+
+  static registerUIToCanvas(element: IUIElement, canvas: UICanvas): void {
+    const preCanvas = element._canvas;
+    if (preCanvas !== canvas) {
+      element._canvas = canvas;
+      if (preCanvas) {
+        const replaced = preCanvas._disorderedElements.deleteByIndex(element._indexInCanvas);
+        replaced && (replaced._indexInCanvas = element._indexInCanvas);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)